firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/firestore.kt

```diff
@@ -28,6 +28,11 @@ expect val emulatorHost: String
 expect val context: Any
 expect fun runTest(test: suspend CoroutineScope.() -> Unit)
 
+/** @return a map extracted from the encoded data. */
+expect fun encodedAsMap(encoded: Any?): Map<String, Any?>
+/** @return pairs as raw encoded data. */
+expect fun Map<String, Any?>.asEncoded(): Any
+
```

Comment:
we have nativeMapOf and nativeAssertEquals  that can be reused instead of adding additional calls

Reply:
@nbransby thanks, I actually missed those. however in order for `nativeMapOf` and `nativeAssertEquals` to be available in `firebase-firestore` tests, the methods shall be moved from the tests folder of the `firebase-common` to a separate module (lets say `firebase-common-test`). shall I make this change?

Reply:
good point - maybe would be easier to do it on a source set level instead of a separate module?

could add a commonTestUtil source set to firebase-common and include in the test source of firebase-firestore with:

```
        val commonTest by getting {
            kotlin.srcDir("../firebase-common/src/commonTestUtil/kotlin")
        }
```

Reply:
the suggested approach created a lot of noise - it required setting up not only common but platform tests. extracted a test-utils module instead